dive_core/tests/perf_metrics_data_test.cpp

```diff
@@ -0,0 +1,185 @@
+/*
+ Copyright 2025 Google LLC
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+#include "dive_core/perf_metrics_data.h"
+#include "dive_core/available_metrics.h"
+#include "gtest/gtest.h"
+#include "gmock/gmock.h"
+
+namespace Dive
+{
+
+void PrintTo(const PerfMetricsRecord& record, std::ostream* os)
+{
+    *os << "PerfMetricsRecord {" << std::endl;
+    *os << "  m_context_id: " << record.m_context_id << "," << std::endl;
+    *os << "  m_process_id: " << record.m_process_id << "," << std::endl;
+    *os << "  m_frame_id: " << record.m_frame_id << "," << std::endl;
+    *os << "  m_cmd_buffer_id: " << record.m_cmd_buffer_id << "," << std::endl;
+    *os << "  m_draw_id: " << record.m_draw_id << "," << std::endl;
+    *os << "  m_draw_label: " << record.m_draw_label << "," << std::endl;
+    *os << "  m_program_id: " << record.m_program_id << "," << std::endl;
+    *os << "  m_draw_type: " << static_cast<int>(record.m_draw_type) << "," << std::endl;
+    *os << "  m_lrz_state: " << static_cast<int>(record.m_lrz_state) << "," << std::endl;
+    *os << "  m_metric_values: [";
+    for (size_t i = 0; i < record.m_metric_values.size(); ++i)
+    {
+        std::visit([os](auto&& arg) { *os << arg; }, record.m_metric_values[i]);
+        if (i < record.m_metric_values.size() - 1)
+        {
+            *os << ", ";
+        }
+    }
+    *os << "]" << std::endl;
+    *os << "}";
+}
+
+namespace
+{
+
+using ::testing::IsEmpty;
+
+MATCHER_P(PerfMetricsRecordEq, expected, "has the correct perf metrics record fields")
+{
+    EXPECT_EQ(arg.m_context_id, expected.m_context_id);
+    EXPECT_EQ(arg.m_process_id, expected.m_process_id);
+    EXPECT_EQ(arg.m_frame_id, expected.m_frame_id);
+    EXPECT_EQ(arg.m_cmd_buffer_id, expected.m_cmd_buffer_id);
+    EXPECT_EQ(arg.m_draw_id, expected.m_draw_id);
+    EXPECT_EQ(arg.m_draw_label, expected.m_draw_label);
+    EXPECT_EQ(arg.m_program_id, expected.m_program_id);
+    EXPECT_EQ(arg.m_draw_type, expected.m_draw_type);
+    EXPECT_EQ(arg.m_lrz_state, expected.m_lrz_state);
+    return true;
+}
+
+TEST(PerfMetricsData, LoadFromCsv)
+{
+    auto available_metrics = Dive::AvailableMetrics::LoadFromCsv(TEST_DATA_DIR
```

Comment:
nit: Can omit the namespace since the test code is in the Dive namespace
```suggestion
    auto available_metrics = AvailableMetrics::LoadFromCsv(TEST_DATA_DIR
```